prow/scripts/cluster-integration/kyma-gke-upgrade.sh

```diff
@@ -309,6 +314,58 @@ function installKyma() {
     "${KYMA_SCRIPTS_DIR}"/is-installed.sh --timeout ${KYMA_INSTALL_TIMEOUT}
 }
 
+function checkTestPodTerminated() {
+    local retry=0
+    local runningPods=0
+    local completedPods=0
+
+    while [ "${retry}" -lt "${UPGRADE_TEST_TIMEOUT}" ]; do
+        for podName in $(kubectl get pods -n "${UPGRADE_TEST_NAMESPACE}" -l "${UPGRADE_TEST_RESOURCE_LABEL}" -o json | jq -sr '.[]|.items[].metadata.name')
+        do
+            runningPods=$((runningPods + 1))
+            phase=$(kubectl get pod "${podName}" -n "${UPGRADE_TEST_NAMESPACE}" -o json | jq '.status.phase')
+
+            if [[ "${phase}" == *"Succeeded"* ]] || [[ "${phase}" == *"Failed"* ]]
+            then
+              echo "Test pod '${podName}' has not terminated, pod phase: ${phase}"
+              completedPods=$((completedPods + 1))
+            fi
+        done
+
+        if [ "${runningPods}" == "${completedPods}" ]
+        then
+            echo "All pods in ${UPGRADE_TEST_NAMESPACE} namespace have not terminated phase"
+            return 1
+        fi
+
+        echo "Not all pods in ${UPGRADE_TEST_NAMESPACE} namespace have not terminated phase. Retry checking."
+        runningPods=0
+        completedPods=0
+        retry=$((retry + 1))
+        sleep 1
+    done
+}
+
+createTestResources() {
+    shout "Create test e2e resources"
+    date
+
+    release=$(basename "${UPGRADE_TEST_PATH}")
```

Comment:
the base path for this `tests/end-to-end/upgrade/chart/upgrade` will be `upgrade` imo it's not a good name for release. The whole path is saying more like this is a chart for e2e upgrade test. If you are using *basename* function then u are lacking that information. IMO u should hardcode the release name and have sth like `e2e-upgrade-test`